metpy/calc/turbulence.py

```diff
@@ -110,3 +110,148 @@ def tke(u, v, w, perturbation=False, axis=-1):
+
+    .. math:: \overline{us} - \overline{u}~\overline{s}
+
+    which is computationally more efficient. This is how the kinematic flux
+    is computed in this function if `perturbation` is False.
+
+    For more information on the subject, please see [2]_.
+
+    References
+    ----------
+    .. [2] Garratt, J.R., 1994: The Atmospheric Boundary Layer. Cambridge
+           University Press, 316 pp.
+    """
+
+    if perturbation:
+        kf = np.mean(vel * b, axis=axis)
+    else:
+        kf = np.mean(vel * b, axis=axis) - \
+             np.mean(vel, axis=axis) * np.mean(b, axis=axis)
+
+    return kf
+
+
+@exporter.export
+def friction_velocity(u, w, v=None, perturbation=False, axis=-1):
+    r"""Compute the friction velocity from the time series of the x, z,
+    and optionally y, velocity components.
+
+    Parameters
+    ----------
+    u : array_like
+        The wind component along the x-axis
+    w : array_like
+        The wind component along the z-axis
+    v : array_like, optional
+        The wind component along the y-axis.
+
+    perturbation : {False, True}, optional
+                   True if the `u`, `w`, and `v` components of wind speed
+                   supplied to the function are perturbation velocities.
+                   If False, perturbation velocities will be calculated by
+                   removing the mean value from each component.
+
+    Returns
+    -------
+    array_like
+        The corresponding friction velocity
+
+    Other Parameters
+    ----------------
+    axis : int
+           The index of the time axis. Default is -1
+
+    See Also
+    --------
+    kinematic_flux : Used to compute the x-component and y-component
+                     vertical kinematic momentum flux(es) used in the
+                     computation of the friction velocity.
+
+    Notes
+    -----
+    The Friction Velocity is computed as:
+
+    .. math:: u_{*} = \sqrt[4]{\left(\overline{u^{\prime}w^{\prime}}\right)^2 +
+                               \left(\overline{v^{\prime}w^{\prime}}\right)^2},
+
+    where :math: \overline{u^{\prime}w^{\prime}} and
+    :math: \overline{v^{\prime}w^{\prime}}
+    are the x-component and y-components of the vertical kinematic momentum
+    flux, respectively. If the optional v component of velocity is not
+    supplied to the function, the computation of the friction velocity is
+    reduced to
+
+    .. math:: u_{*} = \sqrt[4]{\left(\overline{u^{\prime}w^{\prime}}\right)^2}
+
+    For more information on the subject, please see [3]_.
+
+    References
+    ----------
+    .. [3] Garratt, J.R., 1994: The Atmospheric Boundary Layer. Cambridge
+           University Press, 316 pp.
+
+    """
+
+    uw = kinematic_flux(u, w, perturbation=perturbation, axis=axis)
+
+    if v is not None:
+        vw = kinematic_flux(v, w, perturbation=perturbation, axis=axis)
+    else:
+        vw = np.zeros(uw.shape)
+
+    return np.power(uw * uw + vw * vw, 0.25)
```

Comment:
Another optimization point:
```
In [16]: a = np.linspace(0, 100, 50000000)

In [17]: timeit np.sqrt(np.sqrt(a))
1 loops, best of 3: 496 ms per loop

In [18]: timeit np.power(a, 0.25)
1 loops, best of 3: 1.34 s per loop
```
You could eliminate the temps and just do in place sqrt (basing on above):
```
np.sqrt(flux, out=flux)
return np.sqrt(flux)
```
Making sure to comment what you're actually doing.

Library code--hiding the ugly performance optimizations for big arrays!